ports/psoc6/machine_pwm.c

```diff
@@ -0,0 +1,231 @@
+}
+
+static mp_obj_t mp_machine_pwm_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
+    // Check number of arguments
+    mp_arg_check_num(n_args, n_kw, 1, MP_OBJ_FUN_ARGS_MAX, true);
+
+    // Get static peripheral object.
+    machine_pwm_obj_t *self = pwm_obj_alloc();
+    self->pin = pin_addr_by_name(all_args[0]);
+    self->duty_type = DUTY_NOT_SET;
+    self->fz = -1;
+    // self->invert = -1;
+
+    // Initialize PWM
+    cy_rslt_t result = cyhal_pwm_init(&self->pwm_obj, self->pin, NULL);
+
+    // To check whether PWM init is successful
+    if (result != CY_RSLT_SUCCESS) {
+        assert_pin_phy_used(result);
+        mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("PWM initialisation failed with return code %lx !"), result);
+    }
+
+    // Process the remaining parameters.
+    mp_map_t kw_args;
+    mp_map_init_fixed_table(&kw_args, n_kw, all_args + n_args);
+    mp_machine_pwm_init_helper(self, n_args - 1, all_args + 1, &kw_args);
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+static void mp_machine_pwm_deinit(machine_pwm_obj_t *self) {
+    cyhal_pwm_stop(&self->pwm_obj);
+    cyhal_pwm_free(&self->pwm_obj);
+    pwm_obj_free(self);
+}
+
+static mp_obj_t mp_machine_pwm_duty_get_u16(machine_pwm_obj_t *self) {
+    if (self->duty_type == DUTY_NS) {
+        // duty_cycle = pulsewidth(ns)*freq(hz);
+        return mp_obj_new_float(((self->duty) * (self->fz) * 65535) / 1000000000 - 1);
+    } else {
+        return mp_obj_new_float(self->duty);
+    }
+}
+
+// sets the duty cycle as a ratio duty_u16 / 65535.
+static void mp_machine_pwm_duty_set_u16(machine_pwm_obj_t *self, mp_int_t duty_u16) {
+    // Check the value is more than the max value
+    self->duty = duty_u16 > 65535 ? 65535 : duty_u16;
+    self->duty_type = DUTY_U16;
+    pwm_freq_duty_set(&self->pwm_obj, self->fz, (float)(self->duty) / (float)65535); // s conversion of duty_u16 into dutyu16/65535
+}
+
+static mp_obj_t mp_machine_pwm_duty_get_ns(machine_pwm_obj_t *self) {
+    if (self->duty_type == DUTY_U16) {
+        return mp_obj_new_float(((self->duty) * 1000000000) / ((self->fz) * 65535));   // pw (ns) = duty_cycle*10^9/fz
+    } else {
+        return mp_obj_new_float(self->duty);
+    }
+}
+
+// sets the pulse width in nanoseconds
+static void mp_machine_pwm_duty_set_ns(machine_pwm_obj_t *self, mp_int_t duty_ns) {
+    self->duty = duty_ns;
+    self->duty_type = DUTY_NS;
+    pwm_duty_set_ns(&self->pwm_obj, self->fz, duty_ns);
+}
+
+static mp_obj_t mp_machine_pwm_freq_get(machine_pwm_obj_t *self) {
+    return MP_OBJ_NEW_SMALL_INT(self->fz);
+
+}
+
+static void mp_machine_pwm_freq_set(machine_pwm_obj_t *self, mp_int_t freq) {
+    self->fz = freq;
+    pwm_freq_duty_set(&self->pwm_obj, freq, self->duty);
+    if (self->duty_type == DUTY_NS) {
+        self->duty = ((self->duty) * (self->fz) * 65535) / 1000000000;
+        mp_machine_pwm_duty_set_ns(self, self->duty);
+    }
+}
+
+void mod_pwm_deinit() {
```

Comment:
Three of the ports use the `machine_pwm_deinit_all` function name.
![image](https://github.com/user-attachments/assets/1c7a96b1-bc31-4d9e-9d25-7d42b9176abe)

Also `machine_XXX_deinit...`
instead of  `mod_XXX_deinit...` for others func's.

Reply:
That naming convention then should be used for all the rest of the machine classes I assume?

Reply:
not must, but preferably